Add files via upload

Marija1984 opened the pull request:
Hey Mariam, could you please check the appdb.py within the new branch-Merged files?
I think I understand what you have meant in you instructions. What I wanted to do here is to have a rout ('/') that brings us only to the homepage ( in my case it is index.html) because I want people to firstly be able to register. Another route ('/user/new') should take me to the welcoming page (in my case called home.html) and if I go to route('/users') then I get all the "registered" users. I errased the return redirect ('/users') because I don't want to always be redirected to that tabe of users. I have changed POST and GET details within undex.html and home.html. I still get all the same, when I press submit, it takes me to the table of users (which I can also see in my database)..

mariamrf commented:
Hey, the python files are missing? There are only the html files in this PR/no changes to `appdb.py`.

Please use Github Desktop (not the web upload) or the command line tool in order to keep track of the files and the changes :-) Not necessarily for this next commit, but it's an important skill to have and it will help you see the differences you make more easily.

Marija1984 commented:
Ok, I don't know why it's missing, I will send it here. I didn't yet
installed github desktop but I will..

sri, 5. lip 2019. u 12:22 Mariam Maarouf <notifications@github.com> napisao
je:

> Hey, the python files are missing? There are only the html files in this
> PR/no changes to appdb.py.
>
> Please use Github Desktop (not the web upload) or the command line tool in
> order to keep track of the files and the changes :-) Not necessarily for
> this next commit, but it's an important skill to have and it will help you
> see the differences you make more easily.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Marija1984/Platform/pull/3?email_source=notifications&email_token=ALSSLBDJ4BZWBZ7MFTRUVXTPY6HWNA5CNFSM4HTTCMT2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODW7IY5A#issuecomment-499027060>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALSSLBB65B4ZKUW4BVMCQATPY6HWNANCNFSM4HTTCMTQ>
> .
>

from flask import Flask, render_template, request, redirect
from flask_mysqldb import MySQL
import yaml


app = Flask(__name__)

#Configure db
db = yaml.load(open('db.yaml'))
app.config['MYSQL_HOST']=db['mysql_host']
app.config['MYSQL_USER']=db['mysql_user']
app.config['MYSQL_PASSWORD']=db['mysql_password']
app.config['MYSQL_DB']=db['mysql_db']

mysql=MySQL(app)

@app.route('/', methods=['GET','POST'])
def index():
    if request.method=='POST':
        #Fetch form data
        userDetails = request.form
        name=userDetails['name']
        email=userDetails['email']
        cur=mysql.connection.cursor()
        cur.execute("INSERT INTO users(name,email) VALUES (%s,%s)", (name,email))
        mysql.connection.commit()
        cur.close()
        return redirect ('/users')
    return render_template('index.html')

@app.route('/users')
def users():
    cur = mysql.connection.cursor()
    resultValue = cur.execute("SELECT * FROM users")
    if resultValue > 0:
        userDetails = cur.fetchall() 
        return render_template('users.html', userDetails=userDetails)

if __name__ == '__main__':
    app.run(debug=True)

Marija1984 commented:
I don't know why it looks the same and I open it in VSCode it contains extra lines of code. Here is the copy&paste code directly from VSC:


from flask import Flask, render_template, request, redirect
from flask_mysqldb import MySQL
import yaml


app = Flask(__name__)

#Configure db
db = yaml.load(open('db.yaml'))
app.config['MYSQL_HOST']=db['mysql_host']
app.config['MYSQL_USER']=db['mysql_user']
app.config['MYSQL_PASSWORD']=db['mysql_password']
app.config['MYSQL_DB']=db['mysql_db']

mysql=MySQL(app)

@app.route('/user/new', methods=['POST'])
def home():
    if request.method=='POST':
        #Fetch form data
        userDetails = request.form
        name=userDetails['name']
        email=userDetails['email']
        cur=mysql.connection.cursor()
        cur.execute("INSERT INTO users(name,email) VALUES (%s,%s)", (name,email))
        mysql.connection.commit()
        cur.close()
        
        return render_template('home.html')

    
@app.route('/')
def index():
        return render_template('index.html')

@app.route('/users')
def users():
    cur = mysql.connection.cursor()
    resultValue = cur.execute("SELECT * FROM users")
    if resultValue > 0:
        userDetails = cur.fetchall() 
        return render_template('users.html', userDetails=userDetails)

if __name__ == '__main__':
    app.run(debug=True)

Marija1984 commented:
Mariam, I did it! I get the homepage with the first root, I get the login form with the /newuser route and a list of users with/users. I am sooo happy! And also I have managed to push it from github desktop, yeeeey!
Now I want to implement some CSS to make it look a little nicer..

Marija1984 commented:
I have added another route /newuser just to look nicer when one login/register to get this note..

mariamrf commented on appdb.py:
```diff
@@ -15,6 +15,16 @@
 
 @app.route('/', methods=['GET','POST'])
+def home():
+    if request.method=='GET':
```
you don't need `POST` support in your home :) It can be only `GET`.

mariamrf replied:
please take a look at this comment^ you don't need `POST` in `/`. You can remove it from the `methods` and the code itself (`if request.method == 'POST'...`)

mariamrf commented on appdb.py:
```diff
@@ -25,9 +35,10 @@ def index():
         mysql.connection.commit()
         cur.close()
-        return redirect ('/users')
+        return redirect ('/newuser')
```
Nice!

mariamrf commented on db.yaml:
```diff
@@ -0,0 +1,4 @@
+mysql_host: 'localhost'
+mysql_user: 'root'
+mysql_password: 'Marija01101984'
+mysql_db: 'flaskapp'
```
Important note: this file contains secrets (passwords, database credentials, etc). It should NOT be on Github. You can add a note in the README about how to create such a file (or put an example file on Github without the actual values) but remove this file from version control (git).

Marija1984 commented:
Great! I can't wait to get on with building! 🙂

pet, 7. lip 2019. 11:14 Mariam Maarouf <notifications@github.com> je
napisao:

> *@mariamrf* commented on this pull request.
>
> Looks good! Some final comments and we can be done with #1
> <https://github.com/Marija1984/Platform/issues/1>. :)
> ------------------------------
>
> In appdb.py
> <https://github.com/Marija1984/Platform/pull/3#discussion_r291511648>:
>
> > @@ -15,6 +15,16 @@
>  mysql=MySQL(app)
>
>  @app.route('/', methods=['GET','POST'])
> +def home():
> +    if request.method=='GET':
>
> you don't need POST support in your home :) It can be only GET.
> ------------------------------
>
> In appdb.py
> <https://github.com/Marija1984/Platform/pull/3#discussion_r291511804>:
>
> > @@ -25,9 +35,10 @@ def index():
>          cur.execute("INSERT INTO users(name,email) VALUES (%s,%s)", (name,email))
>          mysql.connection.commit()
>          cur.close()
> -        return redirect ('/users')
> +        return redirect ('/newuser')
>
> Nice!
> ------------------------------
>
> In db.yaml
> <https://github.com/Marija1984/Platform/pull/3#discussion_r291512446>:
>
> > @@ -0,0 +1,4 @@
> +mysql_host: 'localhost'
> +mysql_user: 'root'
> +mysql_password: 'Marija01101984'
> +mysql_db: 'flaskapp'
>
> Important note: this file contains secrets (passwords, database
> credentials, etc). It should NOT be on Github. You can add a note in the
> README about how to create such a file (or put an example file on Github
> without the actual values) but remove this file from version control (git).
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Marija1984/Platform/pull/3?email_source=notifications&email_token=ALSSLBCDTRBVJS5ZTZIHLBTPZIRGBA5CNFSM4HTTCMT2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOB24MGRY#pullrequestreview-246989639>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALSSLBA3DAKCP3OVYLV76Z3PZIRGBANCNFSM4HTTCMTQ>
> .
>

mariamrf commented:
Hey Marija, can you please look at/address the comments made in this PR? (would probably be best if you view it via Github)

Marija1984 commented:
hey Mariam, I will do that today, I was travelling home this weekend,
sorry.

uto, 11. lip 2019. 10:24 Mariam Maarouf <notifications@github.com> je
napisao:

> Hey Marija, can you please look at/address the comments made in this PR?
> (would probably be best if you view it via Github)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Marija1984/Platform/pull/3?email_source=notifications&email_token=ALSSLBGVO7N6LURFEEKFPSLPZ5ONTA5CNFSM4HTTCMT2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODXMLB3Q#issuecomment-500740334>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALSSLBCH2UINEMGOT35237DPZ5ONTANCNFSM4HTTCMTQ>
> .
>

mariamrf commented:
No problem :) I was just wondering if you saw them.

Marija1984 commented:
Hey Mariam, I have removed the .yaml file from the repository, but now I
can't run the app directly from the Repo folder, since it doesn't contain
this important file. What am I missing?

uto, 11. lip 2019. u 11:56 Mariam Maarouf <notifications@github.com>
napisao je:

> No problem :) I was just wondering if you saw them.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Marija1984/Platform/pull/3?email_source=notifications&email_token=ALSSLBBEFRAUD5RT7Y736QLPZ5ZFRA5CNFSM4HTTCMT2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODXMTA2I#issuecomment-500772969>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALSSLBD5KPITLQOO3X6VH7DPZ5ZFRANCNFSM4HTTCMTQ>
> .
>

Marija1984 commented:
Now I returned it to the folder to be able to run it, but I won't commit
and push this change..

sri, 12. lip 2019. u 09:40 Marija Šokčević <marijaprodanovic@gmail.com>
napisao je:

> Hey Mariam, I have removed the .yaml file from the repository, but now I
> can't run the app directly from the Repo folder, since it doesn't contain
> this important file. What am I missing?
>
> uto, 11. lip 2019. u 11:56 Mariam Maarouf <notifications@github.com>
> napisao je:
>
>> No problem :) I was just wondering if you saw them.
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/Marija1984/Platform/pull/3?email_source=notifications&email_token=ALSSLBBEFRAUD5RT7Y736QLPZ5ZFRA5CNFSM4HTTCMT2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODXMTA2I#issuecomment-500772969>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/ALSSLBD5KPITLQOO3X6VH7DPZ5ZFRANCNFSM4HTTCMTQ>
>> .
>>
>

Marija1984 commented:
One more thing, when I type the wrong address, it shows me the 404 Not
Found page, is it ok or should I do some additional things?

sri, 12. lip 2019. u 09:42 Marija Šokčević <marijaprodanovic@gmail.com>
napisao je:

> Now I returned it to the folder to be able to run it, but I won't commit
> and push this change..
>
> sri, 12. lip 2019. u 09:40 Marija Šokčević <marijaprodanovic@gmail.com>
> napisao je:
>
>> Hey Mariam, I have removed the .yaml file from the repository, but now I
>> can't run the app directly from the Repo folder, since it doesn't contain
>> this important file. What am I missing?
>>
>> uto, 11. lip 2019. u 11:56 Mariam Maarouf <notifications@github.com>
>> napisao je:
>>
>>> No problem :) I was just wondering if you saw them.
>>>
>>> —
>>> You are receiving this because you authored the thread.
>>> Reply to this email directly, view it on GitHub
>>> <https://github.com/Marija1984/Platform/pull/3?email_source=notifications&email_token=ALSSLBBEFRAUD5RT7Y736QLPZ5ZFRA5CNFSM4HTTCMT2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODXMTA2I#issuecomment-500772969>,
>>> or mute the thread
>>> <https://github.com/notifications/unsubscribe-auth/ALSSLBD5KPITLQOO3X6VH7DPZ5ZFRANCNFSM4HTTCMTQ>
>>> .
>>>
>>

Marija1984 commented:
Added the Error page!

sri, 12. lip 2019. u 10:28 Marija Šokčević <marijaprodanovic@gmail.com>
napisao je:

> One more thing, when I type the wrong address, it shows me the 404 Not
> Found page, is it ok or should I do some additional things?
>
> sri, 12. lip 2019. u 09:42 Marija Šokčević <marijaprodanovic@gmail.com>
> napisao je:
>
>> Now I returned it to the folder to be able to run it, but I won't commit
>> and push this change..
>>
>> sri, 12. lip 2019. u 09:40 Marija Šokčević <marijaprodanovic@gmail.com>
>> napisao je:
>>
>>> Hey Mariam, I have removed the .yaml file from the repository, but now I
>>> can't run the app directly from the Repo folder, since it doesn't contain
>>> this important file. What am I missing?
>>>
>>> uto, 11. lip 2019. u 11:56 Mariam Maarouf <notifications@github.com>
>>> napisao je:
>>>
>>>> No problem :) I was just wondering if you saw them.
>>>>
>>>> —
>>>> You are receiving this because you authored the thread.
>>>> Reply to this email directly, view it on GitHub
>>>> <https://github.com/Marija1984/Platform/pull/3?email_source=notifications&email_token=ALSSLBBEFRAUD5RT7Y736QLPZ5ZFRA5CNFSM4HTTCMT2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODXMTA2I#issuecomment-500772969>,
>>>> or mute the thread
>>>> <https://github.com/notifications/unsubscribe-auth/ALSSLBD5KPITLQOO3X6VH7DPZ5ZFRANCNFSM4HTTCMTQ>
>>>> .
>>>>
>>>

Marija1984 commented:
Changed the login details

sri, 12. lip 2019. u 11:05 Marija Šokčević <marijaprodanovic@gmail.com>
napisao je:

> Added the Error page!
>
> sri, 12. lip 2019. u 10:28 Marija Šokčević <marijaprodanovic@gmail.com>
> napisao je:
>
>> One more thing, when I type the wrong address, it shows me the 404 Not
>> Found page, is it ok or should I do some additional things?
>>
>> sri, 12. lip 2019. u 09:42 Marija Šokčević <marijaprodanovic@gmail.com>
>> napisao je:
>>
>>> Now I returned it to the folder to be able to run it, but I won't commit
>>> and push this change..
>>>
>>> sri, 12. lip 2019. u 09:40 Marija Šokčević <marijaprodanovic@gmail.com>
>>> napisao je:
>>>
>>>> Hey Mariam, I have removed the .yaml file from the repository, but now
>>>> I can't run the app directly from the Repo folder, since it doesn't contain
>>>> this important file. What am I missing?
>>>>
>>>> uto, 11. lip 2019. u 11:56 Mariam Maarouf <notifications@github.com>
>>>> napisao je:
>>>>
>>>>> No problem :) I was just wondering if you saw them.
>>>>>
>>>>> —
>>>>> You are receiving this because you authored the thread.
>>>>> Reply to this email directly, view it on GitHub
>>>>> <https://github.com/Marija1984/Platform/pull/3?email_source=notifications&email_token=ALSSLBBEFRAUD5RT7Y736QLPZ5ZFRA5CNFSM4HTTCMT2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODXMTA2I#issuecomment-500772969>,
>>>>> or mute the thread
>>>>> <https://github.com/notifications/unsubscribe-auth/ALSSLBD5KPITLQOO3X6VH7DPZ5ZFRANCNFSM4HTTCMTQ>
>>>>> .
>>>>>
>>>>

Marija1984 commented:
I need a hint, please! I have a POST method twice in my '/' route and it is
always calling the first one (logically. I want my Login button to lead me
to the login page, which it does, but my Register button to my register
page where they can be registered. I've created another table in my
flaskapp database, called registered. I didn´t want to complicate so I put
the same info required for both actions.. Please help! :)

sri, 12. lip 2019. u 14:13 Marija Šokčević <marijaprodanovic@gmail.com>
napisao je:

> Changed the login details
>
> sri, 12. lip 2019. u 11:05 Marija Šokčević <marijaprodanovic@gmail.com>
> napisao je:
>
>> Added the Error page!
>>
>> sri, 12. lip 2019. u 10:28 Marija Šokčević <marijaprodanovic@gmail.com>
>> napisao je:
>>
>>> One more thing, when I type the wrong address, it shows me the 404 Not
>>> Found page, is it ok or should I do some additional things?
>>>
>>> sri, 12. lip 2019. u 09:42 Marija Šokčević <marijaprodanovic@gmail.com>
>>> napisao je:
>>>
>>>> Now I returned it to the folder to be able to run it, but I won't
>>>> commit and push this change..
>>>>
>>>> sri, 12. lip 2019. u 09:40 Marija Šokčević <marijaprodanovic@gmail.com>
>>>> napisao je:
>>>>
>>>>> Hey Mariam, I have removed the .yaml file from the repository, but now
>>>>> I can't run the app directly from the Repo folder, since it doesn't contain
>>>>> this important file. What am I missing?
>>>>>
>>>>> uto, 11. lip 2019. u 11:56 Mariam Maarouf <notifications@github.com>
>>>>> napisao je:
>>>>>
>>>>>> No problem :) I was just wondering if you saw them.
>>>>>>
>>>>>> —
>>>>>> You are receiving this because you authored the thread.
>>>>>> Reply to this email directly, view it on GitHub
>>>>>> <https://github.com/Marija1984/Platform/pull/3?email_source=notifications&email_token=ALSSLBBEFRAUD5RT7Y736QLPZ5ZFRA5CNFSM4HTTCMT2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODXMTA2I#issuecomment-500772969>,
>>>>>> or mute the thread
>>>>>> <https://github.com/notifications/unsubscribe-auth/ALSSLBD5KPITLQOO3X6VH7DPZ5ZFRANCNFSM4HTTCMTQ>
>>>>>> .
>>>>>>
>>>>>

mariamrf commented:
OK so the 404 thing is normal and expected! I'll comment on the code for your hint :)

mariamrf commented on appdb.py:
```diff
@@ -15,19 +15,32 @@
+    return render_template('success.html')
+
+
+@app.route('/login', methods=['GET','POST'])
```
Maybe this should be called `/register` since we are yet to add the logging in itself.

mariamrf replied:
Okay so maybe some clarification is needed here:

### Registering
Creating a new account by inserting user data into the database. This is what you're doing here in this function (so your page should be `/register` and the `/success` page should say that you've successfully registered).

### Logging in
Enable the user to "enter" their account. What this means is that they enter their details (username, password, etc) and you check if those details match the ones **already in your database**. If they do, you grant them access to their account.
**So what does granting access mean?** In this case it's assigning a variable in the `session` that states that the user is logged in (and that when logged out, is changed) for example.  See the code [here](https://pythonspot.com/login-authentication-with-flask/) for example (read the code to see where this is), and read more about Flask sessions [here](https://www.tutorialspoint.com/flask/flask_sessions.htm).

For now, it is enough to register the users (without logging in or out :)) and the login part can be done separately after we merge this pull request.

mariamrf commented on appdb.py:
```diff
@@ -15,19 +15,32 @@
-        name=userDetails['name']
-        email=userDetails['email']
+        username=userDetails['username']
+        password=userDetails['password']
```
I was going to mention this in the next task because this task only includes having a home page :) But passwords should be hashed + salted. Take a look at [this article](https://itnext.io/how-not-to-store-passwords-4955569e6e84) for example. This is in case there's a breach (a leak of the database by a hacker or technical reason) and also even the developers/administrators of the app itself should never be able to read the passwords.

Using hashing we can do the one thing we want to do: store passwords in an unreadable format while still being able to check if the text entered to log in is the one in the database.

If you have any questions let me know!

mariamrf commented on index.html:
```diff
@@ -0,0 +1,7 @@
+<form method="POST" ACTION="">
```
maybe this page and `home.html`/anything not in the `templates` folder should be deleted now? Are they used?

mariamrf commented on templates/index.html:
```diff
@@ -0,0 +1,7 @@
+<form method="POST" ACTION="">
```
I assume this is for registering? Or is it for logging in? If for logging in, leave it as it is for now.

mariamrf commented on templates/home.html:
```diff
@@ -0,0 +1,14 @@
+     <p>Welcome to Coonect where you can start amazing initiatives!</p>
+   </div>
+  </body>
+</html><form method="POST" ACTION="">
```
this does not have to be a form. What you want to do here is:
- Add a link to register (that will take you to the register page)
- Add a link to login (that will take you to the login page)
you can style the links later with css to make them look like buttons :)

so something like:
```html
<a href="/register">Register</a>
<a href="/login">Login</a>
```
Don't forget to add the corresponding `GET` pages for each! :)

mariamrf commented on templates/users.html:
```diff
@@ -0,0 +1,8 @@
+    {% for user in userDetails %}
+    <tr>
+        <td> {{user[0]}} </td>
+        <td> {{user[1]}} </td>
```
please do not print passwords here :D even after hashing!

Marija1984 commented:
Ok, I was just about changing the login into register when you mentioned
it. Tnx, now it is changed. I left the POST method in home because there I
have a link to register page. Is it wrong? At the moment I have two
buttons, but they lead to a register page.


<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

čet, 13. lip 2019. u 14:45 Mariam Maarouf <notifications@github.com>
napisao je:

> *@mariamrf* commented on this pull request.
>
> I think I've addressed everything but let me know if you are confused/need
> clarification :)
> ------------------------------
>
> In appdb.py
> <https://github.com/Marija1984/Platform/pull/3#discussion_r293349961>:
>
> > @@ -15,6 +15,16 @@
>  mysql=MySQL(app)
>
>  @app.route('/', methods=['GET','POST'])
> +def home():
> +    if request.method=='GET':
>
> please take a look at this comment^ you don't need POST in /. You can
> remove it from the methods and the code itself (if request.method ==
> 'POST'...)
> ------------------------------
>
> In appdb.py
> <https://github.com/Marija1984/Platform/pull/3#discussion_r293350272>:
>
> > @@ -15,19 +15,32 @@
>  mysql=MySQL(app)
>
>  @app.route('/', methods=['GET','POST'])
> +def home():
> +    if request.method=='GET':
> +        return render_template('home.html')
> +    if request.method=='POST':
> +        return render_template('home.html')
> +
> +@app.route('/newuser')
> +def success():
> +    return render_template('success.html')
> +
> +
> +@app.route('/login', methods=['GET','POST'])
>
> Maybe this should be called /register since we are yet to add the logging
> in itself.
> ------------------------------
>
> In appdb.py
> <https://github.com/Marija1984/Platform/pull/3#discussion_r293352218>:
>
> >  def index():
>      if request.method=='POST':
>          #Fetch form data
>          userDetails = request.form
> -        name=userDetails['name']
> -        email=userDetails['email']
> +        username=userDetails['username']
> +        password=userDetails['password']
>
> I was going to mention this in the next task because this task only
> includes having a home page :) But passwords should be hashed + salted.
> Take a look at this article
> <https://itnext.io/how-not-to-store-passwords-4955569e6e84> for example.
> This is in case there's a breach (a leak of the database by a hacker or
> technical reason) and also even the developers/administrators of the app
> itself should never be able to read the passwords.
>
> Using hashing we can do the one thing we want to do: store passwords in an
> unreadable format while still being able to check if the text entered to
> log in is the one in the database.
>
> If you have any questions let me know!
> ------------------------------
>
> In appdb.py
> <https://github.com/Marija1984/Platform/pull/3#discussion_r293355498>:
>
> > @@ -15,19 +15,32 @@
>  mysql=MySQL(app)
>
>  @app.route('/', methods=['GET','POST'])
> +def home():
> +    if request.method=='GET':
> +        return render_template('home.html')
> +    if request.method=='POST':
> +        return render_template('home.html')
> +
> +@app.route('/newuser')
> +def success():
> +    return render_template('success.html')
> +
> +
> +@app.route('/login', methods=['GET','POST'])
>
> Okay so maybe some clarification is needed here:
> Registering
>
> Creating a new account by inserting user data into the database. This is
> what you're doing here in this function (so your page should be /register
> and the /success page should say that you've successfully registered).
> Logging in
>
> Enable the user to "enter" their account. What this means is that they
> enter their details (username, password, etc) and you check if those
> details match the ones *already in your database*. If they do, you grant
> them access to their account.
> *So what does granting access mean?* In this case it's assigning a
> variable in the session that states that the user is logged in (and that
> when logged out, is changed) for example. See the code here
> <https://pythonspot.com/login-authentication-with-flask/> for example
> (read the code to see where this is), and read more about Flask sessions
> here <https://www.tutorialspoint.com/flask/flask_sessions.htm>.
>
> For now, it is enough to register the users (without logging in or out :))
> and the login part can be done separately after we merge this pull request.
> ------------------------------
>
> In index.html
> <https://github.com/Marija1984/Platform/pull/3#discussion_r293356066>:
>
> > @@ -0,0 +1,7 @@
> +<form method="POST" ACTION="">
>
> maybe this page and home.html/anything not in the templates folder should
> be deleted now? Are they used?
> ------------------------------
>
> In templates/index.html
> <https://github.com/Marija1984/Platform/pull/3#discussion_r293356344>:
>
> > @@ -0,0 +1,7 @@
> +<form method="POST" ACTION="">
>
> I assume this is for registering? Or is it for logging in? If for logging
> in, leave it as it is for now.
> ------------------------------
>
> In templates/home.html
> <https://github.com/Marija1984/Platform/pull/3#discussion_r293357497>:
>
> > @@ -0,0 +1,14 @@
> +<!DOCTYPE html>
> +<html>
> +  <body>
> +   <div class="home">
> +     <h1>Dear user!</h1>
> +     <p>Welcome to Coonect where you can start amazing initiatives!</p>
> +   </div>
> +  </body>
> +</html><form method="POST" ACTION="">
>
> this does not have to be a form. What you want to do here is:
>
>    - Add a link to register (that will take you to the register page)
>    - Add a link to login (that will take you to the login page)
>    you can style the links later with css to make them look like buttons
>    :)
>
> so something like:
>
> <a href="/register">Register</a>
> <a href="/login">Login</a>
>
> Don't forget to add the corresponding GET pages for each! :)
> ------------------------------
>
> In templates/users.html
> <https://github.com/Marija1984/Platform/pull/3#discussion_r293357664>:
>
> > @@ -0,0 +1,8 @@
> +<table border = 1>
> +    {% for user in userDetails %}
> +    <tr>
> +        <td> {{user[0]}} </td>
> +        <td> {{user[1]}} </td>
>
> please do not print passwords here :D even after hashing!
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Marija1984/Platform/pull/3?email_source=notifications&email_token=ALSSLBBAMNK2ACK7M6ZK5WDP2I6PPA5CNFSM4HTTCMT2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOB3OC2PY#pullrequestreview-249310527>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALSSLBDBBTD4BLMZ6C3LZRLP2I6PPANCNFSM4HTTCMTQ>
> .
>

<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

Marija1984 commented:
And yes, I moved passwords from users page, there are only usernames now :)

<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.
www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

čet, 13. lip 2019. u 16:26 Marija Šokčević <marijaprodanovic@gmail.com>
napisao je:

> Ok, I was just about changing the login into register when you mentioned
> it. Tnx, now it is changed. I left the POST method in home because there I
> have a link to register page. Is it wrong? At the moment I have two
> buttons, but they lead to a register page.
>
>
>
> <https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail> Virus-free.
> www.avast.com
> <https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
> <#m_-5922843972060580389_DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>
>
> čet, 13. lip 2019. u 14:45 Mariam Maarouf <notifications@github.com>
> napisao je:
>
>> *@mariamrf* commented on this pull request.
>>
>> I think I've addressed everything but let me know if you are
>> confused/need clarification :)
>> ------------------------------
>>
>> In appdb.py
>> <https://github.com/Marija1984/Platform/pull/3#discussion_r293349961>:
>>
>> > @@ -15,6 +15,16 @@
>>  mysql=MySQL(app)
>>
>>  @app.route('/', methods=['GET','POST'])
>> +def home():
>> +    if request.method=='GET':
>>
>> please take a look at this comment^ you don't need POST in /. You can
>> remove it from the methods and the code itself (if request.method ==
>> 'POST'...)
>> ------------------------------
>>
>> In appdb.py
>> <https://github.com/Marija1984/Platform/pull/3#discussion_r293350272>:
>>
>> > @@ -15,19 +15,32 @@
>>  mysql=MySQL(app)
>>
>>  @app.route('/', methods=['GET','POST'])
>> +def home():
>> +    if request.method=='GET':
>> +        return render_template('home.html')
>> +    if request.method=='POST':
>> +        return render_template('home.html')
>> +
>> +@app.route('/newuser')
>> +def success():
>> +    return render_template('success.html')
>> +
>> +
>> +@app.route('/login', methods=['GET','POST'])
>>
>> Maybe this should be called /register since we are yet to add the
>> logging in itself.
>> ------------------------------
>>
>> In appdb.py
>> <https://github.com/Marija1984/Platform/pull/3#discussion_r293352218>:
>>
>> >  def index():
>>      if request.method=='POST':
>>          #Fetch form data
>>          userDetails = request.form
>> -        name=userDetails['name']
>> -        email=userDetails['email']
>> +        username=userDetails['username']
>> +        password=userDetails['password']
>>
>> I was going to mention this in the next task because this task only
>> includes having a home page :) But passwords should be hashed + salted.
>> Take a look at this article
>> <https://itnext.io/how-not-to-store-passwords-4955569e6e84> for example.
>> This is in case there's a breach (a leak of the database by a hacker or
>> technical reason) and also even the developers/administrators of the app
>> itself should never be able to read the passwords.
>>
>> Using hashing we can do the one thing we want to do: store passwords in
>> an unreadable format while still being able to check if the text entered to
>> log in is the one in the database.
>>
>> If you have any questions let me know!
>> ------------------------------
>>
>> In appdb.py
>> <https://github.com/Marija1984/Platform/pull/3#discussion_r293355498>:
>>
>> > @@ -15,19 +15,32 @@
>>  mysql=MySQL(app)
>>
>>  @app.route('/', methods=['GET','POST'])
>> +def home():
>> +    if request.method=='GET':
>> +        return render_template('home.html')
>> +    if request.method=='POST':
>> +        return render_template('home.html')
>> +
>> +@app.route('/newuser')
>> +def success():
>> +    return render_template('success.html')
>> +
>> +
>> +@app.route('/login', methods=['GET','POST'])
>>
>> Okay so maybe some clarification is needed here:
>> Registering
>>
>> Creating a new account by inserting user data into the database. This is
>> what you're doing here in this function (so your page should be /register
>> and the /success page should say that you've successfully registered).
>> Logging in
>>
>> Enable the user to "enter" their account. What this means is that they
>> enter their details (username, password, etc) and you check if those
>> details match the ones *already in your database*. If they do, you grant
>> them access to their account.
>> *So what does granting access mean?* In this case it's assigning a
>> variable in the session that states that the user is logged in (and that
>> when logged out, is changed) for example. See the code here
>> <https://pythonspot.com/login-authentication-with-flask/> for example
>> (read the code to see where this is), and read more about Flask sessions
>> here <https://www.tutorialspoint.com/flask/flask_sessions.htm>.
>>
>> For now, it is enough to register the users (without logging in or out
>> :)) and the login part can be done separately after we merge this pull
>> request.
>> ------------------------------
>>
>> In index.html
>> <https://github.com/Marija1984/Platform/pull/3#discussion_r293356066>:
>>
>> > @@ -0,0 +1,7 @@
>> +<form method="POST" ACTION="">
>>
>> maybe this page and home.html/anything not in the templates folder
>> should be deleted now? Are they used?
>> ------------------------------
>>
>> In templates/index.html
>> <https://github.com/Marija1984/Platform/pull/3#discussion_r293356344>:
>>
>> > @@ -0,0 +1,7 @@
>> +<form method="POST" ACTION="">
>>
>> I assume this is for registering? Or is it for logging in? If for logging
>> in, leave it as it is for now.
>> ------------------------------
>>
>> In templates/home.html
>> <https://github.com/Marija1984/Platform/pull/3#discussion_r293357497>:
>>
>> > @@ -0,0 +1,14 @@
>> +<!DOCTYPE html>
>> +<html>
>> +  <body>
>> +   <div class="home">
>> +     <h1>Dear user!</h1>
>> +     <p>Welcome to Coonect where you can start amazing initiatives!</p>
>> +   </div>
>> +  </body>
>> +</html><form method="POST" ACTION="">
>>
>> this does not have to be a form. What you want to do here is:
>>
>>    - Add a link to register (that will take you to the register page)
>>    - Add a link to login (that will take you to the login page)
>>    you can style the links later with css to make them look like buttons
>>    :)
>>
>> so something like:
>>
>> <a href="/register">Register</a>
>> <a href="/login">Login</a>
>>
>> Don't forget to add the corresponding GET pages for each! :)
>> ------------------------------
>>
>> In templates/users.html
>> <https://github.com/Marija1984/Platform/pull/3#discussion_r293357664>:
>>
>> > @@ -0,0 +1,8 @@
>> +<table border = 1>
>> +    {% for user in userDetails %}
>> +    <tr>
>> +        <td> {{user[0]}} </td>
>> +        <td> {{user[1]}} </td>
>>
>> please do not print passwords here :D even after hashing!
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/Marija1984/Platform/pull/3?email_source=notifications&email_token=ALSSLBBAMNK2ACK7M6ZK5WDP2I6PPA5CNFSM4HTTCMT2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOB3OC2PY#pullrequestreview-249310527>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/ALSSLBDBBTD4BLMZ6C3LZRLP2I6PPANCNFSM4HTTCMTQ>
>> .
>>
>
>
> <https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail> Virus-free.
> www.avast.com
> <https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
> <#m_-5922843972060580389_DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>
>

mariamrf commented:
The `POST` part in `/` should still be removed. I think there's something being confused here, so let me iterate over a few things:

1. **Links**: when you click on a link `/register` for example it submits a `GET` request to `/register`. A link is made by using the [HTML `a` tag](https://www.w3schools.com/tags/tag_a.asp).
2. **Forms**:
--> when you have information you want to enter/give the website, you create a form that uses `POST` to submit this info to some endpoint (e.g. `/register`). For example: registering a user means they enter a username and a password. They are then put in a `POST` request and sent to the server for processing.
--> when you want to get (`GET`) specific information you *may* use a form to provide some special details. This is an advanced use case for forms/for now. You can revisit this concept later.

`POST` should never be used for the sole purpose of getting a resource from the server (e.g. a template or information). See [this resource here](https://www.w3schools.com/tags/ref_httpmethods.asp) for more info :)

Let me know if you have any other questions.

Marija1984 commented:
I did it! Thank you Mariam, god I can be happy when succeed :)

čet, 13. lip 2019. u 17:35 Mariam Maarouf <notifications@github.com>
napisao je:

> The POST part in / should still be removed. I think there's something
> being confused here, so let me iterate over a few things:
>
>    1. *Links*: when you click on a link /register for example it submits
>    a GET request to /register. A link is made by using the HTML a tag
>    <https://www.w3schools.com/tags/tag_a.asp>.
>    2. *Forms*:
>    --> when you have information you want to enter/give the website, you
>    create a form that uses POST to submit this info to some endpoint
>    (e.g. /register). For example: registering a user means they enter a
>    username and a password. They are then put in a POST request and sent
>    to the server for processing.
>    --> when you want to get (GET) specific information you *may* use a
>    form to provide some special details. This is an advanced use case for
>    forms/for now. You can revisit this concept later.
>
> POST should never be used for the sole purpose of getting a resource from
> the server (e.g. a template or information). See this resource here
> <https://www.w3schools.com/tags/ref_httpmethods.asp> for more info :)
>
> Let me know if you have any other questions.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Marija1984/Platform/pull/3?email_source=notifications&email_token=ALSSLBCHQ6HRI6ZWKYKKKRLP2JSMDA5CNFSM4HTTCMT2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODXUC7AY#issuecomment-501755779>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALSSLBHW2RQ7RK6JDJUPMCDP2JSMDANCNFSM4HTTCMTQ>
> .
>

mariamrf commented on appdb.py:
```diff
@@ -14,20 +14,49 @@
     return render_template('index.html')
 
+
+@app.route('/login', methods=['GET','POST']) 
```
we will change this (the functionality) in the next step :)

Marija1984 commented:
Hi.. tried to hash and salt password, but no luck..don't understand how I
should edit the code :(
Going to sleep now, good night..

čet, 13. lip 2019. u 20:58 Mariam Maarouf <notifications@github.com>
napisao je:

> *@mariamrf* approved this pull request.
>
> this is good to go! We'll revisit registering and logging in in the next
> steps :) but for now, good job!
> ------------------------------
>
> In appdb.py
> <https://github.com/Marija1984/Platform/pull/3#discussion_r293530075>:
>
> >      return render_template('index.html')
>
> +
> +@app.route('/login', methods=['GET','POST'])
>
> we will change this (the functionality) in the next step :)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Marija1984/Platform/pull/3?email_source=notifications&email_token=ALSSLBGTRA7UNNK5GT5BLOLP2KKDXA5CNFSM4HTTCMT2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOB3P3V5Q#pullrequestreview-249543414>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALSSLBA4AF4TG6N4D4LXGF3P2KKDXANCNFSM4HTTCMTQ>
> .
>

mariamrf commented:
First, you have to merge this PR and start working on a new branch :) Then tell me: what tutorial or documentation are you looking at/following?

mariamrf commented:
No - merge it don't close it. Now the code isn't on `master`. Please reopen this PR and merge it to master (use Github web interface if that's easier).

Marija1984 commented:
I wanted to merge this branch to master on Github desktop (I don't know
why) and then I seemed to lose all my updated code, the home and index
pages are old, I need to once again go back to your instructions and update
everything.
I hope I will recover this because now the code doesn't work now. I have
only little time before starting to make my presentation.
 I have been reading this https://flask-hashing.readthedocs.io/en/latest/ and
this https://pythonprogramming.net/password-hashing-flask-tutorial/, but
now I don´t have time to solve this one :(




uto, 18. lip 2019. u 10:34 Mariam Maarouf <notifications@github.com>
napisao je:

> First, you have to merge this PR and start working on a new branch :) Then
> tell me: what tutorial or documentation are you looking at/following?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Marija1984/Platform/pull/3?email_source=notifications&email_token=ALSSLBGNRGQB4N6YXKEBMS3P3CMZNA5CNFSM4HTTCMT2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODX5UASI#issuecomment-503005257>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALSSLBDJZXWXUMXU5BFOOZ3P3CMZNANCNFSM4HTTCMTQ>
> .
>

Marija1984 commented:
I don´t know how to reopen it..I wanted to merge it, then something went
wrong. I'll try to fix the code..

uto, 18. lip 2019. u 17:08 Marija Šokčević <marijaprodanovic@gmail.com>
napisao je:

> I wanted to merge this branch to master on Github desktop (I don't know
> why) and then I seemed to lose all my updated code, the home and index
> pages are old, I need to once again go back to your instructions and update
> everything.
> I hope I will recover this because now the code doesn't work now. I have
> only little time before starting to make my presentation.
>  I have been reading this https://flask-hashing.readthedocs.io/en/latest/ and
> this https://pythonprogramming.net/password-hashing-flask-tutorial/, but
> now I don´t have time to solve this one :(
>
>
>
>
> uto, 18. lip 2019. u 10:34 Mariam Maarouf <notifications@github.com>
> napisao je:
>
>> First, you have to merge this PR and start working on a new branch :)
>> Then tell me: what tutorial or documentation are you looking at/following?
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/Marija1984/Platform/pull/3?email_source=notifications&email_token=ALSSLBGNRGQB4N6YXKEBMS3P3CMZNA5CNFSM4HTTCMT2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODX5UASI#issuecomment-503005257>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/ALSSLBDJZXWXUMXU5BFOOZ3P3CMZNANCNFSM4HTTCMTQ>
>> .
>>
>

mariamrf commented:
Are you on Github web or doing this through Github desktop?

Marija1984 commented:
I was doing it on Github desktop.. Now I updated all my code and it works
again. In the meantime I opened the Profile branch.. Once I go to the
GitHub desktop, it asks me do I want to bring my changes to profile, do I
do it?
I'm not sure now how to put the updated code again up to the repository..

uto, 18. lip 2019. u 17:30 Mariam Maarouf <notifications@github.com>
napisao je:

> Are you on Github web or doing this through Github desktop?
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/Marija1984/Platform/pull/3?email_source=notifications&email_token=ALSSLBFX35546252HMXB6U3P3D5PXA5CNFSM4HTTCMT2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODX7AUYI#issuecomment-503188065>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALSSLBDPK54GKO5Q455KYVTP3D5PXANCNFSM4HTTCMTQ>
> .
>

mariamrf commented:
1. Open this link in your browser:
https://github.com/Marija1984/Platform/pull/3

2. Login if you haven't already.
3. Scroll down in the page, you'll find a button that says Reopen Pull Request, click it.
4. Click on Merge Pull Request

Marija1984 commented:
So I won`t be losing my code again? Because in the meantime it has been
lost and now I corrected it..
Sorry for the stupid questions

uto, 18. lip 2019. u 17:41 Mariam Maarouf <notifications@github.com>
napisao je:

>
>    1.
>
>    Open this link in your browser:
>    #3 <https://github.com/Marija1984/Platform/pull/3>
>    2.
>
>    Login if you haven't already.
>    3.
>
>    Scroll down in the page, you'll find a button that says Reopen Pull
>    Request, click it.
>    4.
>
>    Click on Merge Pull Request
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/Marija1984/Platform/pull/3?email_source=notifications&email_token=ALSSLBDH6CKL5Y6P43IVR5DP3D6ZPA5CNFSM4HTTCMT2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODX7B3KA#issuecomment-503193000>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALSSLBA52GQCPYQTG4XZACDP3D6ZPANCNFSM4HTTCMTQ>
> .
>

mariamrf commented:
No you won't lose it, you can merge it now

Marija1984 commented:
Ok, I did it. Still, I don't know where to push files from the GHdesktop
since it asks me if I want to move changes to another branch..

uto, 18. lip 2019. u 17:50 Mariam Maarouf <notifications@github.com>
napisao je:

> No you won't lose it, you can merge it now
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/Marija1984/Platform/pull/3?email_source=notifications&email_token=ALSSLBCJHS6A45QIAGRDMODP3D75DA5CNFSM4HTTCMT2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODX7CZXY#issuecomment-503196895>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALSSLBAYJJHFULHW3FFXN7TP3D75DANCNFSM4HTTCMTQ>
> .
>

mariamrf commented:
If you haven't made any changes after the ones in this PR, remove all current changes, then change the branch to master, and sync it (pull from origin), and then create a new branch and start doing your changes there.

Marija1984 commented:
Ok, now I resolved some pending issues and commited to the master. I have
this new Profile branch, but somehow it contains already some old files,
sould I delete them and start a pull request?

uto, 18. lip 2019. u 17:58 Mariam Maarouf <notifications@github.com>
napisao je:

> If you haven't made any changes after the ones in this PR, remove all
> current changes, then change the branch to master, and sync it (pull from
> origin), and then create a new branch and start doing your changes there.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/Marija1984/Platform/pull/3?email_source=notifications&email_token=ALSSLBFNLG7DTHXD4NILVETP3EA35A5CNFSM4HTTCMT2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODX7DT5I#issuecomment-503200245>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALSSLBBBXECQHLYQJKIQHZLP3EA35ANCNFSM4HTTCMTQ>
> .
>

mariamrf commented:
you should delete the whole branch (Profile) and make a new branch from the current master.

Marija1984 commented:
I did, I created the new branch Profile again from the master. I just don't
have anything new since I don't know from where to start. I have tried
several things, but they don't look good.
So, I don`t have much time left, If I will be able to make, I'd like to
learn how to hash+salt the passwords, how to set conditions to my
username/password and to make the login recognizes the registered user.

sri, 19. lip 2019. u 10:08 Mariam Maarouf <notifications@github.com>
napisao je:

> you should delete the whole branch (Profile) and make a new branch from
> the current master.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/Marija1984/Platform/pull/3?email_source=notifications&email_token=ALSSLBFHZDFB65CFHF2SDUDP3HSOHA5CNFSM4HTTCMT2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODYBBW4Y#issuecomment-503454579>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALSSLBDAASSJ5KTKCFFYPELP3HSOHANCNFSM4HTTCMTQ>
> .
>

mariamrf commented:
What tutorials are you using for the hash+salt? Have you looked at the resources provided in the issue on Github (#4?)

Marija1984 commented:
Hi, I will look at those again later. Now I'm at work and got the task to
do :)

sri, 19. lip 2019. u 12:02 Mariam Maarouf <notifications@github.com>
napisao je:

> What tutorials are you using for the hash+salt? Have you looked at the
> resources provided in the issue on Github (#4
> <https://github.com/Marija1984/Platform/issues/4>?)
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly, view it on GitHub
> <https://github.com/Marija1984/Platform/pull/3?email_source=notifications&email_token=ALSSLBGUUWYX4YZL32BG6NLP3H733A5CNFSM4HTTCMT2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODYBLSYI#issuecomment-503495009>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ALSSLBEIZSEXFLJ7PEOEXXDP3H733ANCNFSM4HTTCMTQ>
> .
>

Marija1984 commented:
Dear Mariam,

I have just commited one last change to master, I did it on purpose since I
dont have time to do additional things. I am at home with my children this
weekend so I dont have the space for exploring new things and getting deep
into problems/issues. I just wanted it to inform you that I will make a
presentation based on what I have till today. Of course, if you have some
important remarks, please tell me.
And thank you for everything, I really learned a lot! :)

Best,
Marija

sri, 19. lip 2019. u 12:05 Marija Šokčević <marijaprodanovic@gmail.com>
napisao je:

> Hi, I will look at those again later. Now I'm at work and got the task to
> do :)
>
> sri, 19. lip 2019. u 12:02 Mariam Maarouf <notifications@github.com>
> napisao je:
>
>> What tutorials are you using for the hash+salt? Have you looked at the
>> resources provided in the issue on Github (#4
>> <https://github.com/Marija1984/Platform/issues/4>?)
>>
>> —
>> You are receiving this because you modified the open/close state.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/Marija1984/Platform/pull/3?email_source=notifications&email_token=ALSSLBGUUWYX4YZL32BG6NLP3H733A5CNFSM4HTTCMT2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODYBLSYI#issuecomment-503495009>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/ALSSLBEIZSEXFLJ7PEOEXXDP3H733ANCNFSM4HTTCMTQ>
>> .
>>
>

mariamrf commented:
Great :) I'll let you know if I have any remarks :)
